sympy/solvers/solveset.py

```diff
@@ -513,7 +513,9 @@ def _solve_radical(f, symbol, solveset_solver):
         result = Union(*[imageset(Lambda(y, g_y), f_y_sols)
                          for g_y in g_y_s])
 
-    return FiniteSet(*[s for s in result if checksol(f, symbol, s) is True])
+    return (result
```

Comment:
I see. The only way to get this output is to define x,y by
```python
x, y = sympy.symbols("x y")
``

If you do
```
x = sympy.Symbol("x", real=True)
y = sympy.Symbol("y", real=True)
```
you'll get the correct answer.